app/controllers/comments_controller.rb

```diff
@@ -24,6 +24,20 @@ def edit
   # POST /comments
   # POST /comments.json
   def create
+    captcha = params['captcha']
+
+    if captcha.nil?
+      flash[:error] = 'Captcha is required!'
+      return redirect_back(fallback_location: root_path)
+    end
+
+    x, y, z, n, result = captcha.values_at('x', 'y', 'z', 'n', 'result').map(&:to_i)
+
+    unless n.between?(100, 999) && n == result && x + y - z == n
```

Comment:
Is there a reason for checking both if n is equal to the result and calculating it?